### ISSUE TITLE
Update notice.md to include all dependency lisences.

### ISSUE THREAD

**Issue by nehashri** [getgauge/gauge#1244]
The notice.md file is quite old and not maintained. This needs to be updated to reflect the latest dependency and changes.
The skel file also includes a notice variable, which is used to generate the notice file while installing Gauge for the first time. This needs to be updated too. It would be ideal if the content for this is copied from the notice.md file itself instead of maintaining it in two places.
Gauge also never updates this file after it is created for the first time. This needs to be changed and it should be updated every time a Gauge update is made to reflect the latest notice.

In summary, things to be done
- Update notice.md file with all latest dependency licenses.
- Generate the notice file for first time installation from notice.md file instead of the variable currently used.
- Every update of Gauge should also update the notice.md file created.
